pkg/istio/authorizationpolicy/controller_test.go

```diff
@@ -136,17 +166,276 @@ func newFakeController(services []*v1.Service, fake bool, stopCh <-chan struct{}
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := newFakeController(tt.inputAthenzDomain, tt.inputService, tt.fake, make(chan struct{}))
+			switch action := tt.item.Operation; action {
+			case model.EventDelete:
+				c.configStoreCache.Create(tt.existingAuthzPolicy)
+				time.Sleep(100 * time.Millisecond)
+				key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(tt.item.Resource)
+				assert.Equal(t, nil, err, "function convert item interface to key should not return error")
+				err = c.sync(key)
+				if err != nil {
+					log.Panicln("controller has sync err: ", err)
+				}
+
+				genAuthzPolicy := c.configStoreCache.Get(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), tt.inputService.Name, tt.inputService.Namespace)
+				if genAuthzPolicy != nil {
+					assert.Errorf(t, fmt.Errorf("authorization policy spec still exists in the cache"), "authorization policy should be deleted after delete action")
+				}
+			// default case refers to EventAdd and EventUpdate action
+			default:
+				time.Sleep(100 * time.Millisecond)
+				key, err := cache.MetaNamespaceKeyFunc(tt.item.Resource)
+				assert.Equal(t, nil, err, "function convert item interface to key should not return error")
+				err = c.sync(key)
+				if err != nil {
+					log.Panicln("controller has sync err: ", err)
+				}
+
+				// Updated Spec should be equal
+				genAuthzPolicy := c.configStoreCache.Get(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), tt.inputService.Name, tt.inputService.Namespace)
+				if genAuthzPolicy != nil {
+					assert.Equal(t, tt.expectedAuthzPolicy.Spec, genAuthzPolicy.Spec, "created authorization policy spec should be equal")
+					// except creation timestamp and revision, configMeta field should be same
+					assert.Equal(t, tt.expectedAuthzPolicy.ConfigMeta.Namespace, genAuthzPolicy.ConfigMeta.Namespace, "namespace should be equal")
+					assert.Equal(t, tt.expectedAuthzPolicy.ConfigMeta.Type, genAuthzPolicy.ConfigMeta.Type, "resource type should be equal")
+					assert.Equal(t, tt.expectedAuthzPolicy.ConfigMeta.Name, genAuthzPolicy.ConfigMeta.Name, "resource name should be equal")
+					assert.Equal(t, tt.expectedAuthzPolicy.ConfigMeta.Group, genAuthzPolicy.ConfigMeta.Group, "apiGroup should be equal")
+					assert.Equal(t, tt.expectedAuthzPolicy.ConfigMeta.Version, genAuthzPolicy.ConfigMeta.Version, "apiVersion should be equal")
+				} else {
+					// generated spec returns nil, compare with empty config
+					assert.Equal(t, tt.expectedAuthzPolicy, model.Config{}, "generated authorization policy should be nil")
+				}
+			}
+		})
+	}
+}
+
+func TestSyncAthenzDomain(t *testing.T) {
+	tests := []struct {
+		name                string
+		expectedAuthzPolicy model.Config
+		fake                bool
+		item                Item
+	}{
+		{
+			name:                "update existing authz policy spec when there is athenz domain crd update",
```

Comment:
Can we also add a test case to check if athenz domain create and delete are working correctly?